docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,197 @@
-# Testing your contracts
+# Testing contracts
 
-\<New guide, TODO>
+After [compiling your contracts](./compile-contracts.md), the next step is to write some tests to verify that they work as intended. This guide explains our recommended approach for testing contracts in Hardhat. It relies on [ethers](https://docs.ethers.io/v5/) to connect to the Hardhat Network and [Mocha](https://mochajs.org/) and [Chai](https://www.chaijs.com/) to write the tests. It also uses our custom [Chai matchers](/hardhat-chai-matchers) and our [Hardhat Network Helpers](/hardhat-network-helpers) to make it easier to write contracts-related tests.
```

Comment:
Maybe mention that they come installed with hardhat toolbox, and you would have them if you followed the other guides

Reply:
Added a sentence in ec8735dfe79fedc6173deec80d7ea28517b4a382